ChaosMod/Effects/db/Player/PlayerRandomVehSeat.cpp

```diff
@@ -14,15 +14,21 @@ static void OnStart()
 
 		std::vector<Vehicle> vehs;
 
+		float groundZ;
 		for (Vehicle veh : GetAllVehs())
 		{
-			vehs.push_back(veh);
+			Vector3 vehPos = GET_ENTITY_COORDS(veh, false);
+			bool useGroundZ = GET_GROUND_Z_FOR_3D_COORD(vehPos.x, vehPos.y, vehPos.z, &groundZ, false, false);
+			bool vehCollision = HAS_COLLISION_LOADED_AROUND_ENTITY(veh);
```

Comment:
Combine these bools please. The other check doesn’t have to be executed if the first check already failed
Same thing for the other effect

Reply:
I'm not an actual dev I must admit but I gave it a shot (see new commit) 👍